airbyte-integrations/bases/connector-acceptance-test/Dockerfile

```
@@ -1,39 +1,27 @@
-FROM python:3.9.11-alpine3.15 as base
+FROM python:3.10.12 as base
 
-# build and load all requirements
-FROM base as builder
-WORKDIR /airbyte/integration_code
 
-# upgrade pip to the latest version
-RUN apk --no-cache upgrade \
+ENV ACCEPTANCE_TEST_DOCKER_CONTAINER 1
+ENV DOCKER_VERSION = "24.0.2"
+
+RUN apt-get update \
     && pip install --upgrade pip \
-    && apk --no-cache add tzdata build-base
+    && apt-get install tzdata bash curl
 
-COPY setup.py ./
-# install necessary packages to a temporary folder
-RUN python setup.py egg_info
-RUN pip install --prefix=/install -r *.egg-info/requires.txt
+# Docker is required for the dagger in docker use case.
+RUN curl -fsSL https://get.docker.com | sh
 
-# build a clean environment
-FROM base
-WORKDIR /airbyte/connector_acceptance_test
 
-# copy all loaded and built libraries to a pure basic image
-COPY --from=builder /install /usr/local
-# add default timezone settings
-COPY --from=builder /usr/share/zoneinfo/Etc/UTC /etc/localtime
-RUN echo "Etc/UTC" > /etc/timezone
-# Bash is installed for more convenient debugging.
-RUN apk --no-cache add bash
 
-ENV ACCEPTANCE_TEST_DOCKER_CONTAINER 1
+RUN pip install poetry==1.5.1
+RUN poetry config virtualenvs.create false 
+RUN echo "Etc/UTC" > /etc/timezone
 
-# copy payload code only
-COPY pytest.ini setup.py ./
-COPY connector_acceptance_test ./connector_acceptance_test
-RUN pip install .
+WORKDIR /app
+COPY . /app
+RUN poetry install --only main --no-cache --no-interaction --no-ansi
```

We should copy the pyproject.toml and install ahead of bringing the code along to get a build speed improvement
```suggestion
WORKDIR /app
COPY pyproject.toml /app
COPY poetry.lock /app
RUN poetry install --only main --no-cache --no-interaction --no-ansi
COPY . /app
```

---

@bnchrch I think we'd only get this speed improvement if we'd install without the root package first to cache the deps:
```
WORKDIR /app
COPY pyproject.toml /app
COPY poetry.lock /app
RUN poetry install --no-root --only main --no-interaction --no-ansi
COPY . /app
RUN poetry install --only main --no-cache --no-interaction --no-ansi
```

Otherwise `poetry install` will try to install the root package without its code which would fail isn't it?

---

Hmm I may be misinterpretting what you mean by root package

What I understand that to mean is the airbyte-integrations/bases/connector-acceptance-test/ folder.

And If that the case then running `poetry install` with only the `pyproject.toml`and `poetry.lock` files and nothing else in the container should pass.

The only case Im think that would fail is if we had relative local imports